pkgs/default.nix

```diff
@@ -6,115 +6,66 @@ let
   modules = self.modules;
   revstring_long = self.rev or "dirty";
   revstring = builtins.substring 0 7 revstring_long;
-  all-modules = builtins.fromJSON (builtins.readFile ../modules.json);
 
-  upgrade-maps = import ./upgrade-maps {
-    inherit pkgs;
-  };
-
-  bundle-locked-fn = { modulesLocks }: pkgs.callPackage ./bundle-locked {
-    inherit self modulesLocks upgrade-maps;
-  };
+  dev-module-ids = [ "python-3.10" "nodejs-18" "nodejs-20" "docker" "replit" "replit-rtld-loader" ];
 
   mkPhonyOCI = pkgs.callPackage ./mk-phony-oci { ztoc-rs = self.inputs.ztoc-rs.packages.x86_64-linux.default; };
 
-  mkPhonyOCIs = { moduleIds ? null }: pkgs.callPackage ./mk-phony-ocis {
-    inherit mkPhonyOCI revstring;
-    modulesLocks = import ./filter-modules-locks {
-      inherit pkgs moduleIds;
-    };
-  };
+  bundle-fn = pkgs.callPackage ./bundle { inherit self; };
 
   bundle-squashfs-fn = { moduleIds ? null, diskName ? "disk.raw" }:
-    let
-      modulesLocks = import ./filter-modules-locks {
-        inherit pkgs upgrade-maps;
-        inherit moduleIds;
-      };
-    in
     pkgs.callPackage ./bundle-image {
-      bundle-locked = bundle-locked-fn {
-        inherit modulesLocks;
-      };
+      bundle = bundle-fn { inherit moduleIds; };
       inherit revstring diskName;
     };
 
 in
 rec {
-  inherit upgrade-maps;
-
   default = moduleit;
   moduleit = pkgs.callPackage ./moduleit { };
 
-  bundle = pkgs.linkFarm "nixmodules-bundle-${revstring}" (
-    mapAttrsToList (name: value: { inherit name; path = value; }) modules
-  );
+  activeRegistry = pkgs.callPackage ./registry {
+    modules = self.activeModules;
+  };
 
-  rev = pkgs.writeText "rev" revstring;
+  registry = pkgs.callPackage ./registry {
+    inherit modules;
+  };
 
-  rev_long = pkgs.writeText "rev_long" revstring_long;
+  bundle = bundle-fn { };
 
-  bundle-locked = bundle-locked-fn {
-    modulesLocks = import ./filter-modules-locks {
-      inherit pkgs upgrade-maps;
-    };
+  custom-bundle = bundle-fn {
+    moduleIds = dev-module-ids;
   };
 
-  custom-bundle-locked = bundle-locked-fn {
-    modulesLocks = import ./filter-modules-locks {
-      inherit pkgs upgrade-maps;
-      moduleIds = [ "python-3.10" "nodejs-20" ];
-    };
-  };
+  rev = pkgs.writeText "rev" revstring;
 
-  inherit (bundle-locked) active-modules registry;
+  rev_long = pkgs.writeText "rev_long" revstring_long;
 
   bundle-image = bundle-squashfs-fn { };
 
+  # For prod use: builds the Nixmodules disk image
   bundle-image-tarball = pkgs.callPackage ./bundle-image-tarball { inherit bundle-image revstring; };
 
-  dev-module-ids = [ "python-3.10" "nodejs-18" "nodejs-20" "docker" "replit" "replit-rtld-loader" ];
-
+  # For dev use: builds the shared Nixmodules disk
   bundle-squashfs = bundle-squashfs-fn {
     moduleIds = dev-module-ids;
     diskName = "disk.sqsh";
   };
 
+  # For dev use: with goval's `make custom-nixmodules-disk`
```

Comment:
I wonder if we don't really need this anymore since you can develop nix modules in Repls.

Reply:
Still need the disk setup to deploy the registry.json and modules.json and just e2e setup.